### PR TITLE
Random ids gen cleanup

### LINKED ISSUE #642: Clean logic in RandomIdsGenerator.
Context: https://github.com/open-telemetry/opentelemetry-java/pull/640/files#r339639316

### PR DESCRIPTION
**Pull request by breedx-nr**
This resolves #642 

Note: ThreadLocalRandom was added in android api 21 and we need to support down to 14?  Removed to comment as a result.

### CODE REVIEW

File: sdk/src/main/java/io/opentelemetry/sdk/trace/RandomIdsGenerator.java [hunk 1]
```diff
@@ -32,21 +30,22 @@
...
 
   @Override
   public TraceId generateTraceId() {
-    long idHi;
-    long idLo;
-    do {
-      idHi = random.nextLong();
-      idLo = random.nextLong();
-    } while (idHi == INVALID_ID && idLo == INVALID_ID);
+    long idHi = generateValidId();
```
Comment (bogdandrutu) [inline sdk/src/main/java/io/opentelemetry/sdk/trace/RandomIdsGenerator.java]: This is not equivalent with the previous logic I think. The previous logic was that for trace ID the whole id should not be 0, individual parts can be. Am I missing something?

Comment (breedx-nr) [reply sdk/src/main/java/io/opentelemetry/sdk/trace/RandomIdsGenerator.java]: Correct, this is a small change in logic.  Is this change problematic, or just surprising, or both?

Comment (carlosalberto) [reply sdk/src/main/java/io/opentelemetry/sdk/trace/RandomIdsGenerator.java]: I'd rather keep both as zero for sanity purposes. There was been discussion in other places about what to do for vendors that to not support a 64 bit ids, and whether they should use the most significant or least significant bits, so this *could* impact them.

File: sdk/src/main/java/io/opentelemetry/sdk/trace/RandomIdsGenerator.java [hunk 2]
```diff
@@ -22,8 +22,6 @@
 
 final class RandomIdsGenerator implements IdsGenerator {
   private static final long INVALID_ID = 0;
-  // TODO change with ThreadLocal version
```
Comment (carlosalberto) [inline sdk/src/main/java/io/opentelemetry/sdk/trace/RandomIdsGenerator.java]: We still don't have this one solved, no?

File: sdk/src/main/java/io/opentelemetry/sdk/trace/RandomIdsGenerator.java (hunk 1 above)
Comment (breedx-nr) [reply sdk/src/main/java/io/opentelemetry/sdk/trace/RandomIdsGenerator.java]: Just to be clear, you mean that you'd prefer that either one may be zero, like the former logic?

File: sdk/src/main/java/io/opentelemetry/sdk/trace/RandomIdsGenerator.java (hunk 2 above)
Comment (breedx-nr) [reply sdk/src/main/java/io/opentelemetry/sdk/trace/RandomIdsGenerator.java]: Nope, because Android.

**Comment by breedx-nr** [comment]
This has been sitting open/stalled for a while, so I'm going to close it.  I guess its a bit off from what was expected or what we wanted to do...so let's clarify the issue and have another go at it another time.

**Pull request closed without merging**